theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-toolbar-contribution.tsx

```diff
@@ -20,24 +21,56 @@ export class TraceViewerToolbarContribution implements TabBarToolbarContribution
     @inject(CommandRegistry)
     protected readonly commands: CommandRegistry;
 
-    private annotationsMap: Map<string, boolean> = new Map<string, boolean>();
-    private onAnnotationsFetchedSignal = (annotationsList: string[]) => this.doHandleAnnotationsFetchedSignal(annotationsList);
+    private traceViewerAnnotationsMap: Map<string, Map<string, Map<string, boolean>>> = new Map<string, Map<string, Map<string, boolean>>>();
+    private traceViewerMarkerSets: Map<string, Map<MarkerSet, boolean>> = new Map<string, Map<MarkerSet, boolean>>();
+    private onAnnotationsFetchedSignal = (payload: { traceViewerId: string, outputId: string, annotationCategories: string[] }) => this.doHandleAnnotationsFetchedSignal(payload);
     protected readonly onAnnotationsChangedEmitter = new Emitter<void>();
     protected readonly onAnnotationsChangedEvent = this.onAnnotationsChangedEmitter.event;
+    private onMarkerSetsFetchedSignal = (payload: { traceViewerId: string, markerSets: MarkerSet[] }) => this.doHandleMarkerSetsFetchedSignal(payload);
+    protected readonly onMarkerSetsChangedEmitter = new Emitter<void>();
+    protected readonly onMarkerSetsChangedEvent = this.onMarkerSetsChangedEmitter.event;
 
     @postConstruct()
     protected init(): void {
         signalManager().on(Signals.ANNOTATIONS_FETCHED, this.onAnnotationsFetchedSignal);
+        signalManager().on(Signals.MARKERSETS_FETCHED, this.onMarkerSetsFetchedSignal);
     }
 
-    private doHandleAnnotationsFetchedSignal(annotationsList: string[]) {
+    private doHandleAnnotationsFetchedSignal(payload: { traceViewerId: string, outputId: string, annotationCategories: string[] }) {
+
+        const traceViewerId = payload.traceViewerId;
+        const outputId = payload.outputId;
+        const annotationsList = payload.annotationCategories;
+        const outputAnnotationsMap: Map<string, Map<string, boolean>> = new Map<string, Map<string, boolean>>();
+        const annotationsMap: Map<string, boolean> = new Map<string, boolean>();
+
         annotationsList.forEach(annotation => {
-            if (!this.annotationsMap.has(annotation)) {
-                this.annotationsMap.set(annotation, true);
+            if (!annotationsMap.has(annotation)) {
```

Comment:
This is the reason why opening a new output doesn't preserve existing category state, because you are working on a brand new map here instead of the current traceViewerId's map.